src/main/scala/org/apache/spark/sql/kinesis/KinesisContinousReader.scala

```diff
@@ -179,6 +180,11 @@ class KinesisContinuousReader(
           return (latestShards.nonEmpty &&
             (openShards(latestShards).toSet -- knownOpenShards).size > 0)
         }
+        catch {
+               case NonFatal(error) =>
```

Comment:
Any particular reason we have to catch the exception here? Do we also need to make similar changes in KinesisSource.scala?

Reply:
Not really necessary, I started fixing the warnings (uncaught `try`) that the compiler threw up and stopped after fixing the first warning. Still intend to fix all warnings and replace deprecated APIs with new ones.

Reply:
Thanks for this. Somehow we missed fixing the warning earlier.